packages/core/src/util/index.js

```diff
@@ -5,19 +5,74 @@
 
 'use strict';
 
-module.exports = {
-  applicationUnderMonitoring: require('./applicationUnderMonitoring'),
-  atMostOnce: require('./atMostOnce'),
-  clone: require('./clone'),
-  compression: require('./compression'),
-  ensureNestedObjectExists: require('./ensureNestedObjectExists'),
-  excludedFromInstrumentation: require('./excludedFromInstrumentation'),
-  hasThePackageBeenInitializedTooLate: require('./initializedTooLateHeuristic'),
-  normalizeConfig: require('./normalizeConfig'),
-  propertySizes: require('./propertySizes'),
-  requireHook: require('./requireHook'),
-  slidingWindow: require('./slidingWindow'),
-  stackTrace: require('./stackTrace'),
-  iitmHook: require('./iitmHook'),
-  hook: require('./hook')
+const applicationUnderMonitoring = require('./applicationUnderMonitoring');
+const clone = require('./clone');
+const compression = require('./compression');
+const ensureNestedObjectExists = require('./ensureNestedObjectExists');
+const excludedFromInstrumentation = require('./excludedFromInstrumentation');
+const hasThePackageBeenInitializedTooLate = require('./initializedTooLateHeuristic');
+const normalizeConfig = require('./normalizeConfig');
+const propertySizes = require('./propertySizes');
+const requireHook = require('./requireHook');
+const slidingWindow = require('./slidingWindow');
+const stackTrace = require('./stackTrace');
+const iitmHook = require('./iitmHook');
+const hook = require('./hook');
+const getPreloadFlags = require('./getPreloadFlags');
+
+/** @type {import('@instana/core/src/core').GenericLogger} */
+let logger;
+
+/**
+ * @param {import('@instana/core/src/util/normalizeConfig').InstanaConfig} config
+ */
+exports.init = function init(config) {
+  logger = config.logger;
+  hasThePackageBeenInitializedTooLate.init(config);
+  applicationUnderMonitoring.init(config);
+  requireHook.init(config);
+  getPreloadFlags.init(config);
+  iitmHook.init(config);
 };
+
+exports.applicationUnderMonitoring = applicationUnderMonitoring;
+
+/**
+ * Make sure that a function is only ever called once. This is useful to maintain
+ * the contract that a callback should only ever be called once.
+ *
+ * Any violations against this contract will be logged for further analysis.
+ *
+ * @param {String} name The name of the callback to make debugging easier.
+ * @param {(...args: *) => *} cb The callback to execute at most once.
+ * @return {(...args: *) => *} A wrapped function which will forward the first call to `cb`
+ *   and log any successive calls.
+ */
+exports.atMostOnce = function atMostOnce(name, cb) {
```

Comment:
Move the content of the file "atMostOnce" into this file for now, because I could not find a nice way to add an init fn to this little helper.

Reply:
> Index.js serves as the interface

100%. But its out of the scope for this PR to refactor it.